### PR TITLE
Refactoring StorefrontConfigurableProductBasicInfoTest

### LINKED ISSUE #31844: [Issue] Refactoring StorefrontConfigurableProductBasicInfoTest
This issue is automatically created based on existing pull request: magento/magento2#31827: Refactoring StorefrontConfigurableProductBasicInfoTest

---------

### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
Run the test in the local environment and verified that it run successfully.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by kate-kyzyma**
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
Run the test in the local environment and verified that it run successfully.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#31844: Refactoring StorefrontConfigurableProductBasicInfoTest

### CODE REVIEW

**Comment by kate-kyzyma** [comment]
run all tests

**Comment by kate-kyzyma** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Semantic Version Checker, Functional Tests EE, Functional Tests CE

File: app/code/Magento/ConfigurableProduct/Test/Mftf/Test/StorefrontConfigurableProductDetailsTest/StorefrontConfigurableProductBasicInfoTest.xml [hunk 1]
```diff
@@ -38,17 +38,35 @@
...
-        <see userInput="1.00" selector="{{StorefrontProductInfoMainSection.productPrice}}" stepKey="seeProductPrice"/>
-        <see userInput="{{colorProductAttribute.default_label}}" selector="{{StorefrontProductInfoMainSection.productAttributeTitle1}}" stepKey="seeProductAttributeTitle"/>
-        <see selector="{{StorefrontProductInfoMainSection.productAttributeTitle1}}" userInput="{{colorProductAttribute.default_label}}" stepKey="seeColorAttributeName1"/>
-        <see userInput="{{colorProductAttribute1.name}}" selector="{{StorefrontProductInfoMainSection.productAttributeOptions1}}" stepKey="seeInDropDown1"/>
-        <see userInput="{{colorProductAttribute2.name}}" selector="{{StorefrontProductInfoMainSection.productAttributeOptions1}}" stepKey="seeInDropDown2"/>
-        <see userInput="{{colorProductAttribute3.name}}" selector="{{StorefrontProductInfoMainSection.productAttributeOptions1}}" stepKey="seeInDropDown3"/>
+        <actionGroup ref="AssertStorefrontProductStockStatusOnProductPageActionGroup" stepKey="seeProductStatusInStoreFront">
```
Comment (eduard13) [inline app/code/Magento/ConfigurableProduct/Test/Mftf/Test/StorefrontConfigurableProductDetailsTest/StorefrontConfigurableProductBasicInfoTest.xml]: Same here.
```suggestion
        <actionGroup ref="AssertStorefrontProductStockStatusOnProductPageActionGroup" stepKey="seeProductStockStatus">
```

File: app/code/Magento/ConfigurableProduct/Test/Mftf/Test/StorefrontConfigurableProductDetailsTest/StorefrontConfigurableProductBasicInfoTest.xml [hunk 2]
```diff
@@ -38,17 +38,35 @@
         </after>
 
         <!-- Verify configurable product details in storefront product view -->
-        <amOnPage url="{{_defaultProduct.urlKey}}.html" stepKey="amOnConfigurableProductPage"/>
-        <waitForPageLoad stepKey="wait"/>
-        <see userInput="{{_defaultProduct.name}}" selector="{{StorefrontProductInfoMainSection.productName}}" stepKey="seeProductName"/>
-        <see userInput="{{_defaultProduct.sku}}" selector="{{StorefrontProductInfoMainSection.productSku}}" stepKey="seeProductSku"/>
+        <actionGroup ref="StorefrontOpenProductPageActionGroup" stepKey="openStorefrontProductPage">
```
Comment (eduard13) [inline app/code/Magento/ConfigurableProduct/Test/Mftf/Test/StorefrontConfigurableProductDetailsTest/StorefrontConfigurableProductBasicInfoTest.xml]: We should preserve the step keys to be a backward compatible change.
```suggestion
        <actionGroup ref="StorefrontOpenProductPageActionGroup" stepKey="amOnConfigurableProductPage">
```

File: app/code/Magento/ConfigurableProduct/Test/Mftf/ActionGroup/AssertStorefrontProductAtttributeOptionVisibleActionGroup.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AssertStorefrontProductAtttributeOptionVisibleActionGroup">
```
Comment (eduard13) [inline app/code/Magento/ConfigurableProduct/Test/Mftf/ActionGroup/AssertStorefrontProductAtttributeOptionVisibleActionGroup.xml]: A typo here. Please rename the file and adjust the test.
```suggestion
    <actionGroup name="AssertStorefrontProductAttributeOptionVisibleActionGroup">
```

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @kate-kyzyma, thank you for the MFTF improvement. Could you please review the below comments?

**Comment by gabrieldagama** [comment]
@magento create issue

File: app/code/Magento/ConfigurableProduct/Test/Mftf/Test/StorefrontConfigurableProductDetailsTest/StorefrontConfigurableProductBasicInfoTest.xml (hunk 1 above)
Comment (engcom-Hotel) [reply app/code/Magento/ConfigurableProduct/Test/Mftf/Test/StorefrontConfigurableProductDetailsTest/StorefrontConfigurableProductBasicInfoTest.xml]: Has been renamed

File: app/code/Magento/ConfigurableProduct/Test/Mftf/Test/StorefrontConfigurableProductDetailsTest/StorefrontConfigurableProductBasicInfoTest.xml (hunk 2 above)
Comment (engcom-Hotel) [reply app/code/Magento/ConfigurableProduct/Test/Mftf/Test/StorefrontConfigurableProductDetailsTest/StorefrontConfigurableProductBasicInfoTest.xml]: Has been renamed

File: app/code/Magento/ConfigurableProduct/Test/Mftf/ActionGroup/AssertStorefrontProductAtttributeOptionVisibleActionGroup.xml (hunk 3 above)
Comment (engcom-Hotel) [reply app/code/Magento/ConfigurableProduct/Test/Mftf/ActionGroup/AssertStorefrontProductAtttributeOptionVisibleActionGroup.xml]: Has been renamed

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Integration Tests, Sample Data Tests B2B, WebAPI Tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Review by eduard13 — APPROVED** [review]
Thank you @engcom-Hotel for assistance here.

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8707](https://jira.corp.magento.com/browse/ENGCOM-8707) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
QA not applicable

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Pull request merged**